components/sync15/src/client.rs

```diff
@@ -25,16 +27,38 @@ pub struct Sync15StorageClientInit {
     pub tokenserver_url: Url,
 }
 
+fn get_response_timestamp(resp: &Response) -> error::Result<ServerTimestamp> {
+    Ok(resp
+        .headers()
+        .get(X_LAST_MODIFIED)
+        .and_then(|v| v.to_str().ok())
+        .and_then(|s| ServerTimestamp::from_str(s).ok())
+        .ok_or_else(|| ErrorKind::MissingServerTimestamp)?)
+}
+
+/// A TimestampedResponse is used as a wrapper for any non-collection response
+/// we may want to update with an x-if-unmodified-since header to ensure we are
+/// overwriting what we think we are.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
```

Comment:
Is there a chance we might accidentally upload a `TimestampedResponse` as a BSO payload, instead of its record?